examples/bind-shopping.proposed.amp.html

```diff
@@ -0,0 +1,271 @@
+              "image": "./shirts/black.jpeg",
+              "sizes": ["S", "L"]
+            },
+            {
+              "id": "1",
+              "name": "34 BROWN",
+              "image": "./shirts/brown.jpeg",
+              "sizes": ["M", "L"]
+            },
+            {
+              "id": "2",
+              "name": "03 GRAY",
+              "image": "./shirts/gray.jpeg",
+              "sizes": ["S"]
+            },
+            {
+              "id": "3",
+              "name": "69 NAVY",
+              "image": "./shirts/navy.jpeg",
+              "sizes": ["S", "L"]
+            },
+            {
+              "id": "4",
+              "name": "19 WINE",
+              "image": "./shirts/wine.jpeg",
+              "sizes": []
+            },
+            {
+              "id": "5",
+              "name": "61 BLUE",
+              "image": "./shirts/blue.jpeg",
+              "sizes": ["L"]
+            },
+            {
+              "id": "6",
+              "name": "59 DARK GREEN",
+              "image": "./shirts/dark-green.jpeg",
+              "sizes": ["S"]
+            },
+            {
+              "id": "7",
+              "name": "02 LIGHT GRAY",
+              "image": "./shirts/light-gray.jpeg",
+              "sizes": ["S", "M", "L"]
+            },
+            {
+              "id": "8",
+              "name": "00 WHITE",
+              "image": "./shirts/white.jpeg",
+              "sizes": []
+            }
+          ]
+        }
+      </script>
+    </amp-bind-dataset>
+
+    <amp-carousel slide=0 id="carousel-1" width=1 height=1 layout=responsive type=slides controls
+        bind-attr="{slide: selected}"
+        on="slidechange:setState({selected: ampEventData})">
+
+      <amp-img src="./shirts/black.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/brown.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/gray.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/navy.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/wine.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/blue.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/dark-green.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/light-gray.jpeg" width=1 height=1 layout=responsive></amp-img>
+      <amp-img src="./shirts/white.jpeg" width=1 height=1 layout=responsive></amp-img>
+    </amp-carousel>
+
+    <amp-img class="color" width=355 height=50 src="https://placehold.it/355x50?text=COLOR%20???"
+        bind-src="https://placehold.it/300x400?text=COLOR%20{{items[selected].name}}">
+    </amp-img>
+
+    <div class="thumbnails">
+      <amp-img src="./shirts/black.jpeg" width=50 height=50 layout=fixed
+          on="tap:setState({selected: 0})"
+          bind-attr="{selected: selected === 0}"
```

Comment:
I originally suggested `bind-<attr>` as a convenient alternative. Dima raised the concern that `bind-<attr>` could be less performant in the DOM scan. However, if the set of bindable attributes is small in practice, you may be right that `bind-<attr>` could be a better way to go. 

I like `[attr]`. The "bind" name itself is also still up for discussion.